design/authenticators-status/authn_status_oidc.md

```diff
@@ -0,0 +1,74 @@
+# Feature Overview
+
+***As a*** Conjur operator\
+***I'd like to*** know that the OIDC authenticator is configured correctly\
+***So that*** I can complete the configuration properly if it's incomplete (or remove
+it if it's not needed).
+
+This feature lets the person who configures the OIDC authenticator to get an immediate feedback
+on the configuration, before any user needs to run an authentication request.
+
+As mentioned in the general status check implementation details, in addition to the general validations,
+some of the authenticators will have additional validations based on their specific requirements.  
+
+***Note***: At this time, we will only be implementing the OIDC status check. 
+Later implementations of authenticator status checks will be done incrementally. 
+Therefore, if the authenticator does not have the status check implemented, then a 503 error will 
+be returned. 
+
+# Implementation Details
+
+As mentioned in the general status check implementation details, we will add the following for the 
+OIDC status check:
+
+1. A `Status` CommandClass in the structure `Authentication::AuthnOidc::Status`
+1. A `status` method to its existing authenticator class (i.e. `Authentication::AuthnOidc::Authenticator`)
+which will call the new CommandClass, as follows: 
+ 
+ ```
+ def status
+   Authentication::AuthnOidc::Status.new.(
+     <input for status check>
+   )
+ end
+ ``` 
+
+The new CommandClass `Authentication::AuthnOidc::Status` will consist of a
+`call` method that will perform the following checks:
+
+- The following variables exist and have value
+    - `provider-uri`
+    - `id-token-user-property`
+- The group `conjur/authn-oidc/<service-id>/users` exists and has `read` & `authenticate` 
```

Comment:
@orenbm we need read? not only authenticate? blackout here

Reply:
I am actually not sure that this is a valid check...
Today we provide freedom to our users to provide any name they want. they can create a layer called conjur/authn-oidc/<service-id>/myfavoritehosts` with `read` & `authenticate`permissions and that is fine.
is there a prossibility to test it the other way around - meaning that other then the oidc authn owner there are  users or hosts that have `read` & `authenticate`permissions?
if not maybe we should drop this check

Reply:
> @orenbm we need read? not only authenticate? blackout here
We do need the privileges of `authenticate` AND `read` because with `read`, the user that is a member of the group will be able to view the permitted webservices (in the UI for example).

Reply:
regarding the name - the group name above is defined in the oidc policy. it has the policy id and the name `users`.

as you can see in the docs (https://docs-staging.conjur.org/staging_a/en/Content/OIDC/OIDC.htm?tocpath=Integrations%7CAuthenticators%7C_____1) we later tell the customers to add groups/users to that group. this is the best practice to grant permissions to `webservice-users` & `secrets-users`.

Reply:
after chatting in live - i understand what you mean @InbalZilberman . i removed this check and explained in the docs why we don't check it